util/stream.js

```diff
@@ -116,6 +116,7 @@ module.exports = function(log) {
 		return initDependency(self, [stream], derive, recover)
 	}
 	function combine(fn, streams) {
+		if (streams.length > streams.filter(valid).length) throw new Error("Ensure that each item passed to m.prop.combine/m.prop.merge is a stream")
```

Comment:
This is pretty much what I initially thought to do, but I was gun-shy 😛 

This still isn't going to fix the loop problem, though - the render function will keep calling `combine` and it will always throw the same error.